git-rebase--interactive.sh

```diff
@@ -127,54 +127,6 @@ init_revisions_and_shortrevisions () {
 	fi
 }
 
-complete_action() {
-	test -s "$todo" || echo noop >> "$todo"
-	test -z "$autosquash" || git rebase--helper --rearrange-squash || exit
-	test -n "$cmd" && git rebase--helper --add-exec-commands "$cmd"
-
-	todocount=$(git stripspace --strip-comments <"$todo" | wc -l)
-	todocount=${todocount##* }
-
-cat >>"$todo" <<EOF
-
-$comment_char $(eval_ngettext \
-	"Rebase \$shortrevisions onto \$shortonto (\$todocount command)" \
-	"Rebase \$shortrevisions onto \$shortonto (\$todocount commands)" \
-	"$todocount")
-EOF
-	git rebase--helper --append-todo-help ${keep_empty:+--keep-empty}
-
-	has_action "$todo" ||
-		return 2
```

Comment:
Why isn't there code for the above 2 lines in the function you wrote?

Reply:
Return code 2 is a special case for interactive rebase. It tells git-rebase.sh to apply the autostash, to remove the state dir and to die with the message “nothing to do” ([see here](https://github.com/git/git/blob/master/git-rebase.sh#L215-L220)). I decided to rewrite this case in C, so instead of `return 2`, there is [this](https://github.com/git/git/pull/513/files/d77c4414835ec158c87d863e849f1c5f9c322a6d..0b4b1c0fb3cd9bbc80f1914ebbebf2a81b38886d#diff-79231f0693f84f3951daeea17065aad9R4615).

After the call to `--append-todo-help`, I noticed that the todo-list has at least one action: if the todo file is empty, then complete_action() writes a noop there. So I removed this specific check from my code.

Reply:
It also reminds me that I forgot to remove the handling of return code 2 in git-rebase.sh.

Reply:
Oh no, it would break git-rebase--preserve-merges.sh.

Reply:
Ok, please explain all that in the commit message.